src/app/services/article.service.ts

```diff
@@ -179,9 +134,6 @@ export class ArticleService {
     const newArt = article;
     newArt.authorId = author.uid;
     newArt.articleId = articleId;
```

Comment:
We definitely still want all 3 to exist on any new article, but feel free to just return the set operation instead of the whole outcome work-around Ian put together.
```suggestion
    newArt.articleId = articleId;
    newArt.commentCount = 0;
    newArt.version = 1;
    newArt.viewCount = 0; //will be used soon and this helps avoid large scale data updates.
```